### DISCUSSION TITLE
reference content file with space in filename

### DISCUSSION THREAD

**Discussion by cycomanic** [executablebooks/meta#654] [Q&A]
I am compiling a book composed of chapters which are files with spaces in their  filenames. I seem not to be able to make this work. 

I've tried:
```
[text](./filename with space.md)
[text](./filename%20with%20space.md)
[text](./filename\ with\ space.md)
[text]("./filename with space.md)
{doc}`text <./filename with space.md>`
{doc}`text <./filename%20with%20space.md>`
{doc}`text <./filename\ with\ space.md>`
```
but none of these seem to work. Is there a way of making this work? This issue really only comes up because jupyter names notebooks after the title, which often does contain spaces. Any other solution, except for putting labels on every chapter?

**Comment by choldgraf** [comment]
I think you've actually uncovered a shortcoming in myst-parser - I opened this issue to track it: https://github.com/executablebooks/MyST-Parser/issues/518

In the meantime, you should be able to get away with `` {doc}`text <filename with space>` ``. If you're directly using `doc` then you should omit the file extension.

**Reply by chrisjsewell** [reply]
Yeh but it's not valid common mark, so we just can't support it
